cwl/src/test/scala/wdl4s/cwl/CwlWorkflowWomSpec.scala

```diff
@@ -0,0 +1,301 @@
+          case Valid(wom) =>
+            wom.entryPoint match {
+              case workflow: TaskDefinition =>
+
+                workflow.inputs shouldBe Set(RequiredInputDefinition("message", WdlStringType))
+
+                workflow.graph.map {
+                  graph =>
+                    graph.nodes.collect { case gin: GraphInputNode => gin.name } should be(Set("echo.message"))
+                    graph.nodes collect { case cn: CallNode => cn.name } should be(Set("echo"))
+
+                    graph.nodes.collectFirst { case echo: CallNode if echo.name == "echo" => echo }.get.
+                      upstream shouldBe Set(RequiredGraphInputNode("echo.message", WdlStringType))
+                }
+
+              case _ => fail("not a workflow")
+            }
+          case i@Invalid(_) => fail(s"Invalid: $i")
+        }
+      case Right((wth: Any, _)) => fail(s"parsed unexpected type $wth")
+      case Left(error) => fail(s"did not parse!  $error")
+    }
+  }
+
+  "Cwl for 1st workflow" should "convert to WOM" in {
+    val firstWorkflow =
+      s"""
+cwlVersion: "v1.0"
+class: "Workflow"
+inputs:
+  - type: "string"
+    id: "file:///home/dan/wdl4s/r.cwl#ex"
+  - type: "File"
+    id: "file:///home/dan/wdl4s/r.cwl#inp"
+outputs:
+  - type: "File"
+    outputSource: "file:///home/dan/wdl4s/r.cwl#compile/classfile"
+    id: "file:///home/dan/wdl4s/r.cwl#classout"
+steps:
+  - run: "cwl/src/test/resources/arguments.cwl"
+    in:
+      -
+        source: "file:///home/dan/wdl4s/r.cwl#untar/example_out"
+        id: "file:///home/dan/wdl4s/r.cwl#compile/src"
+    out:
+      - "file:///home/dan/wdl4s/r.cwl#compile/classfile"
+    id: "file:///home/dan/wdl4s/r.cwl#compile"
+  - run: "cwl/src/test/resources/tar-param.cwl"
+    in:
+      -
+        source: "file:///home/dan/wdl4s/r.cwl#ex"
+        id: "file:///home/dan/wdl4s/r.cwl#untar/extractfile"
+      -
+        source: "file:///home/dan/wdl4s/r.cwl#inp"
+        id: "file:///home/dan/wdl4s/r.cwl#untar/tarfile"
+    out:
+      - "file:///home/dan/wdl4s/r.cwl#untar/example_out"
+    id: "file:///home/dan/wdl4s/r.cwl#untar"
+id: "file:///home/dan/wdl4s/r.cwl"
+name: "file:///home/dan/wdl4s/r.cwl"
+
+""".stripMargin
+
+    import CwlCodecs._
+
+
+    decodeCwl(firstWorkflow) match {
+      case Right((workflow: Workflow, nameToFile)) =>
+        workflow.womExecutable(nameToFile) match {
+          case Valid(ex) => validateWom(ex)
+          case e => fail(s"executable was not created $e")
+        }
+      case Right((wth: Any, _)) => fail(s"Parsed unexpected CwlFile subtype $wth")
+      case Left(error) => fail(s"did not parse!  $error")
+    }
+
+    def validateWom(ex: Executable) = {
+      ex match {
+        case Executable(wf: WorkflowDefinition) =>
+          val nodes = wf.innerGraph.nodes
+
+          nodes collect {
+            case gin: GraphInputNode => gin.name
+          } should be(Set("file:///home/dan/wdl4s/r.cwl#ex", "file:///home/dan/wdl4s/r.cwl#inp"))
```

Comment:
Otherwise, our workflow graphs are going to look like this:

![outfile](https://user-images.githubusercontent.com/13006282/29042976-3b06682e-7b87-11e7-9787-1ba573c2a12e.png)